src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -63,6 +77,37 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
             codeDocument.SetSyntaxTree(rewrittenSyntaxTree);
         }
 
+        private string ComputeNamespace(RazorCodeDocument codeDocument)
+        {
+            var rootNamespace = codeDocument.GetCodeGenerationOptions().RootNamespace;
+            var relativePath = codeDocument.Source.RelativePath;
```

Comment:
This is basically a copy-paste from `ComponentDocumentClassifierPass`. I'll try to share this logic if possible.